tests/src/RunBenchmarks/RunBenchmarks.cs

```diff
@@ -0,0 +1,1049 @@
+        public Controls Controls = new Controls()
+        {
+            NumberOfRunsPerBenchmark = 1,
+            DoWarmUpRun = false,
+            DoVerbose = false,
+            Runner = "",
+            DoDebugBenchmark = false,
+            DoListBenchmarks = false,
+            DoListBenchmarkSuites = false,
+            DoListBenchmarkTagSets = false,
+            DoRun = true,
+            ComplusVersion = "",
+            SuiteName = "",
+            BenchmarksRootDirectory = "",
+            BenchmarkXmlFileName = "benchmarks.xml",
+            IncludeBenchmarkList = new List<string>(),
+            ExcludeBenchmarkList = new List<string>(),
+            IncludeTagList = new List<string>(),
+            ExcludeTagList = new List<string>()
+        };
+
+        public Dictionary<string, BenchmarkSuite>
+            BenchmarkSuiteTable = new Dictionary<string, BenchmarkSuite>();
+        public Dictionary<string, BenchmarkTagSet>
+            BenchmarkTagSetTable = new Dictionary<string, BenchmarkTagSet>();
+        public List<Benchmark>
+            BenchmarkList = new List<Benchmark>();
+        public List<Benchmark>
+            SelectedBenchmarkList = new List<Benchmark>();
+        public List<Results>
+            ResultsList = new List<Results>();
+
+        public int NumberOfBenchmarksRun;
+        public int Failures = 0;
+
+        public Char[] ListSeparatorCharSet = new Char[] { ',', ' ' };
+        public char[] DirectorySeparatorCharSet = new char[] { '/', '\\' };
+
+        // Main driver for benchmark system.
+
+        public static int Main(string[] args)
+        {
+            Console.WriteLine("RyuJIT Benchmark System");
+            try
+            {
+                BenchmarkSystem benchmarkSystem = new BenchmarkSystem();
+                benchmarkSystem.ProcessCommandLine(args);
+                benchmarkSystem.BuildBenchmarksList();
+                benchmarkSystem.SelectBenchmarks();
+                benchmarkSystem.RunBenchmarks();
+                benchmarkSystem.ReportResults();
+            }
+            catch (Exception exception)
+            {
+                //  Need to find portable Environment.Exit()
+                if (exception.Message == "Exit")
+                {
+                    return 0;
+                }
+                Console.WriteLine("{0}", exception.ToString());
+                return -1;
+            }
+            return 0;
+        }
+
+        // Command line processor.
+
+        public void ProcessCommandLine(string[] args)
+        {
+            Controls controls = Controls;
+            try
+            {
+                for (int i = 0; i < args.Length;)
+                {
+                    string arg = args[i++];
+                    string benchmark;
+                    string[] tags;
+                    string runner;
+
+                    switch (arg)
+                    {
+                        case "-n":
+                            arg = args[i++];
```

Comment:
you'll obviously throw an exception if someone passes "-n" without a subsequent argument.